### PR TITLE
allow specifying files

### LINKED ISSUE #3: Allow to specify input file(s)
... so that `go run . in.txt` does the same as `head in.txt`.

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #3

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -4,13 +4,42 @@ package main
...
+			file.Close()
+		}
+	} else {
+		countAndPrint(os.Stdin)
+	}
+}
+
+func countAndPrint(input *os.File) {
+	if len(flag.Args()) > 1 {
+		fmt.Printf("==> %v <==\n", input.Name())
```
Comment (Al2Klimov) [inline main.go]: `%v`?

File: main.go [hunk 2]
```diff
@@ -4,13 +4,42 @@ package main
...
+			file, fileErr := os.Open(i)
+			if fileErr != nil {
+				fmt.Fprintln(os.Stderr, fileErr)
+				os.Exit(1)
+			}
+
+			countAndPrint(file)
+
+			if j != len(flag.Args())-1 {
+				fmt.Printf("\n")
```
Comment (Al2Klimov) [inline main.go]: Why Print**f**?

File: main.go [hunk 3]
```diff
@@ -4,13 +4,42 @@ package main
...
 	"io"
 	"os"
 )
 
 func main() {
-	buf := bufio.NewReader(os.Stdin)
+	flag.Parse()
+
+	if len(flag.Args()) > 0 {
+		for j, i := range flag.Args() {
```
Comment (Al2Klimov) [inline main.go]: What are i and j?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/allow-specifying-files-3`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol, test <, one file and multiple ones.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**in.txt:**
```
one
two
three
four
five
six
seven
eight
nine
ten
eleven
twelve
```
**in2.txt:**
```
eins
zwei
drei
vier
fünf
sechs
sieben
acht
neun
zehn
elf
zwölf
```

**Case 1 (`<` one file):**

**Terminal:** `go run . < in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```
**Terminal:** `head < in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```

**Case 2 (no `<`, one file):**

**Terminal:** `go run . in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```
**Terminal:** `head in.txt`
**Output:**
```
one
two
three
four
five
six
seven
eight
nine
ten
```

**Case 3 (no `<`, two files):**

**Terminal:** `go run . in.txt in2.txt`
**Output:**
```
==> in.txt <==
one
two
three
four
five
six
seven
eight
nine
ten

==> in2.txt <==
eins
zwei
drei
vier
fünf
sechs
sieben
acht
neun
zehn
```
**Terminal:** `head in.txt in2.txt`
**Output:**
```
==> in.txt <==
one
two
three
four
five
six
seven
eight
nine
ten

==> in2.txt <==
eins
zwei
drei
vier
fünf
sechs
sieben
acht
neun
zehn
```

**Conclusion:**
All outputs are equal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**